### ISSUE TITLE
Make service name optional for HTTP + Thrift

### ISSUE THREAD

**Issue by prashantv** [yarpc/yab#48]
Right now, we require the service name (which is intended to be the TChannel service name), but this isn't required for HTTP

**Comment by kriskowal** [comment]
@prashantv Can you clarify this issue? What enforces the service name requirement where it should be relaxed. What accepts HTTP Thrift that would not use the service name? Is this under the "Apache Thrift HTTP service compatibility" heading?

**Comment by prashantv** [comment]
Yeah this is under the normal Apache Thrift+HTTP, which is what most external Thrift users are probably using.
